unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,176 @@
+import uuid
+from unittest import mock
+
+import pytest
+from redis import StrictRedis
+import d3a_api_client
+from d3a_api_client.redis_client_base import RedisClientBase, RedisAPIException
+from unittest.mock import MagicMock
+from unittest.mock import Mock
+from unittest.mock import patch
+
+area_id = str(uuid.uuid4())
+area_uid = None
+d3a_api_client.redis_client_base.StrictRedis = MagicMock(spec=StrictRedis)
+d3a_api_client.redis_client_base.StrictRedis.publish = MagicMock(spec=StrictRedis)
+d3a_api_client.redis_client_base.StrictRedis.pubsub.psubscribe = MagicMock()
+
+
+@pytest.fixture
+def redis_client_auto_register():
+    return RedisClientBase(area_id=area_id, autoregister=False)
+
+
+class TestRedisClientBase:
+    @classmethod
+    def setup_class(cls):
+        cls.obj = RedisClientBase(area_id=area_id, autoregister=False)
```

Comment:
This is a mixture of unittest and pytest practices. Why dont you use the fixture e.g. in the `test_constructor_registry_success` ?

Reply:
I agree with Hannes' comment. If you want to set the fixture scope to be "class" scope you can use [fixture scopes](https://docs.pytest.org/en/latest/how-to/fixtures.html#fixture-scopes) like this:
```python
@pytest.fixture(scope="class")
def your_function():
   ...
```

Reply:
Moreover, I would recommend you to use more specific and explanatory variable names. For example, you could replace `obj` with `redis_client`.